network/tests/cleanup.rs

```diff
@@ -77,52 +77,79 @@ async fn check_node_cleanup() {
     #[global_allocator]
     static PEAK_ALLOC: PeakAlloc = PeakAlloc;
 
+    const NUM_CONNS: usize = 256;
```

Comment:
I'd just add a short comment here explaining that it's the connection-related leaks this test is checking for. Also, since the connection count is lower now, does the test take a long time to complete? Otherwise it might be a good idea to un-`#[ignore]` it.